programs/drift/src/instructions/lp_pool.rs

```diff
@@ -0,0 +1,1554 @@
+        ctx.accounts.amm_cache.load_zc()?;
+    let expected_amm_pda = &Pubkey::create_program_address(
+        &[
+            AMM_POSITIONS_CACHE.as_ref(),
+            amm_cache.fixed.bump.to_le_bytes().as_ref(),
+        ],
+        &crate::ID,
+    )
+    .map_err(|_| ErrorCode::InvalidPDA)?;
+    validate!(
+        amm_cache_key.eq(expected_amm_pda),
+        ErrorCode::InvalidPDA,
+        "Amm cache PDA does not match expected PDA"
+    )?;
+
+    let mut aum: u128 = 0;
+    let mut crypto_delta = 0_i128;
+    let mut oldest_slot = u64::MAX;
+    let mut derivative_groups: BTreeMap<u16, Vec<u16>> = BTreeMap::new();
+    for i in 0..lp_pool.constituents as usize {
+        let constituent = constituent_map.get_ref(&(i as u16))?;
+        if slot.saturating_sub(constituent.last_oracle_slot)
+            > MAX_CONSTITUENT_ORACLE_SLOT_STALENESS_FOR_AUM
+        {
+            msg!(
+                "Constituent {} oracle slot is too stale: {}, current slot: {}",
+                constituent.constituent_index,
+                constituent.last_oracle_slot,
+                slot
+            );
+            return Err(ErrorCode::ConstituentOracleStale.into());
+        }
+
+        if constituent.constituent_derivative_index >= 0 && constituent.derivative_weight != 0 {
+            if !derivative_groups.contains_key(&(constituent.constituent_derivative_index as u16)) {
+                derivative_groups.insert(
+                    constituent.constituent_derivative_index as u16,
+                    vec![constituent.constituent_index],
+                );
+            } else {
+                derivative_groups
+                    .get_mut(&(constituent.constituent_derivative_index as u16))
+                    .unwrap()
+                    .push(constituent.constituent_index);
+            }
+        }
+
+        let spot_market = spot_market_map.get_ref(&constituent.spot_market_index)?;
+
+        let oracle_slot = constituent.last_oracle_slot;
+
+        if oracle_slot < oldest_slot {
+            oldest_slot = oracle_slot;
+        }
+
+        let (numerator_scale, denominator_scale) = if spot_market.decimals > 6 {
+            (10_i128.pow(spot_market.decimals - 6), 1)
+        } else {
+            (1, 10_i128.pow(6 - spot_market.decimals))
+        };
+
+        let constituent_aum = constituent
+            .get_full_balance(&spot_market)?
+            .safe_mul(numerator_scale)?
+            .safe_div(denominator_scale)?
+            .safe_mul(constituent.last_oracle_price as i128)?
+            .safe_div(PRICE_PRECISION_I128)?
+            .max(0);
+        msg!(
+            "constituent: {}, aum: {}, deriv index: {}",
+            constituent.constituent_index,
+            constituent_aum,
+            constituent.constituent_derivative_index
+        );
+        if constituent.constituent_index != lp_pool.usdc_consituent_index
+            && constituent.constituent_derivative_index != lp_pool.usdc_consituent_index as i16
```

Comment:
if usdc_consituent_index is 0, will constituent.constituent_derivative_index != lp_pool.usdc_consituent_inde cause issues?